client/valor/client.py

```diff
@@ -192,10 +192,9 @@ def _requests_wrapper(
         method_name: str,
         endpoint: str,
         ignore_auth: bool = False,
-        max_retries_on_timeout=2,
-        initial_timeout: float = 2,
+        timeout: Optional[float] = 2,
```

Comment:
move optional parameter to the bottom of the parameter list?

Reply:
This is pretty nitty. 

Could you reference a standard that says this?

A value of None disables the timeout and disables the parameters below it so this seems to be the more intuitive ordering.

Reply:
this is a general requirement for many other programming languages (e.g., JS, but not Python) that take positional arguments. if the user passes positional arguments instead of kw arguments and your `timeout` isn't at the end, then you're essentially making `timeout` mandatory instead of optional.

Reply:
But none of the other arguments are positional... they are all kw args.

Reply:
> But none of the other arguments are positional... they are all kw args

all python functions allow for positional arguments, we just mostly use kwargs on the backend. imagine a future Valor dev who isn't as careful but only uses positional args (and not any kwargs): make their life easier by making all of these args optional, instead of just `timeout`

Reply:
this is one of the reasons you'll people use `-1` to mean "don't use a timeout" instead of `None`

Reply:
The issue of people new to the codebase mistakenly setting kw parameters in the client backend seems like a bit of a reach. 

I just pushed a commit that disables vague `args` for all the requests wrappers in the `client.py` file as they are all somewhat vague anyway. 

The GET, POST, PUT and DELETE wrappers now take endpoint as a positional and all other params must be kwargs.

Reply:
> this is one of the reasons you'll see people use 0 or -1 to mean "don't use a timeout" instead of None

I agree, but the requests library accepts `float | None` for its timeout kwarg.

Reply:
sure, but requests also allows you to pass `0` to mean "no timeout". this would be way less confusing then the way it's written now, right? then you just have to check if `timeout > 0`

> See requests src they even put a non-optional kwarg after...

"non-optional" means "has no default"; all the args after `timeout` in `requests` have a default, so this statement isn't true.

all of your args after `timeout` here all have a defaults, so (I believe) they are also all technically optional. you're specifying `timeout` as `Optional`, but not the others, which is confusing: might as well say they're all `Optional`

Reply:
Would changing it to Union[float, None] work better for you? Because the value is not optional, it has a default of 2 and can be set to None to disable it.

Reply:
Also i cant find it in the docs that setting it to zero disables it.

Reply:
> all of your args after timeout here all have a defaults, so (I believe) they are also all technically optional. you're specifying timeout as Optional, but not the others, which is confusing: might as well say they're all Optional

Yes but in python `typing.Optional[float]` is an alias of `typing.Union[float, None]` so it doesnt make sense at all to tag those optional b/c `None` is NOT a valid value for them. If the user doesnt specify them they use their defaults.

Reply:
Also we are including generic kwargs so the optional isnt even the last kwarg...

but if changing 
```
        ignore_auth: bool = False,
        timeout: Optional[float] = 2,
        max_retries=2,
        exponential_backoff: int = 2,
        **kwargs,
```
to
```
        ignore_auth: bool = False,
        max_retries=2,
        exponential_backoff: int = 2,
        timeout: Optional[float] = 2,
        **kwargs,
```
unblocks this for you then ill make the change

Reply:
> Would changing it to Union[float, None] work better for you? Because the value is not optional, it has a default of 2 and can be set to None to disable it.

that would be less confusing, thanks. the best strategy here IMO is just not to use `None` to mean "no timeout", but you seem pretty set on `None`.

[here is a prior issue](https://github.com/psf/requests/issues/1615) from 2013 which states that `timeout=0` means "no timeout".

Reply:
That issue contains the following 

>  they are thoroughly distinct values in this context. I would be okay with seeing timeout=0 and immediately raising a Timeout exception before doing anything. Since we don't return a response (or request) this won't affect the API.

Reply:
if you read the bottom of the issue, it says this was fixed and merged back in 2013. the issue is helpful because it communicates the devs intent.

Reply:
There was no code associated with that issue. The linked urllib3 pr at the end also shows `timeout=0` raises a `ReadTimeoutError`.

Reply:
Also should of thought of this earlier but you can literally see it doesnt work by trying it

```
>>> requests.get("https://google.com")
<Response [200]>
>>> requests.get("https://google.com", timeout=0)
ValueError: Attempted to set connect timeout to 0, but the timeout cannot be set to a value less than or equal to 0.
```

Reply:
good catch. you could always check for 0 or -1 instead of checking for `None` to avoid this whole issue. you seem pretty set on using `None`, and I'm not that invested in this change